middleware/queue_test.go

```diff
@@ -0,0 +1,235 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestQueueWithConfig(t *testing.T) {
+	e := echo.New()
+
+	handler := func(c echo.Context) error {
+		pts := c.Get("procTime")
+		procTime, ok := pts.(int)
+		if !ok {
+			return c.NoContent(http.StatusInternalServerError)
+		}
+
+		time.Sleep(time.Duration(procTime) * time.Millisecond)
+
+		return c.NoContent(http.StatusOK)
+	}
+
+	mw := QueueWithConfig(QueueConfig{
+		QueueSize:     2,
+		Workers:       1,
+		QueueTimeout:  200 * time.Millisecond,
+		WorkerTimeout: 100 * time.Millisecond,
+	})
+
+	testCases := []struct {
+		procTime int // in Milliseconds
+	}{
+		{50},
+		{95},
+		{95},
+		{95},
+		{95},
+		{120},
```

Comment:
The 120 and 250 procTime are too near to the Queue and Worker timeouts. Maybe this could lead to random failures when this test are executed. Please run them several times and set values to this procTime that don't generate random failures.

Reply:
Unfortunately, I ran them 100 times before committing but I've adjusted the test cases to try to make it as reliable as possible.